### PR TITLE
Add CodecContext to allow getting the session inside of a Codec

### PR DESCRIPTION
**Pull request by jdonkervliet**
Cherry-picked commit from @Paulomart: https://github.com/Paulomart/network/commit/bd32ed91b8b2b78360490c98ecd10f5ff73dcd56

### CODE REVIEW

File: src/main/java/com/flowpowered/network/pipeline/MessageDecoder.java [hunk 1]
```diff
@@ -39,13 +40,17 @@
...
     private final MessageHandler messageHandler;
+    private CodecContext context;
 
     public MessageDecoder(final MessageHandler handler) {
         this.messageHandler = handler;
     }
 
     @Override
     protected void decode(ChannelHandlerContext ctx, ByteBuf buf, List<Object> out) throws Exception {
+    	if (context == null && messageHandler.getSession() != null) {
```
Comment (aramperes) [inline src/main/java/com/flowpowered/network/pipeline/MessageDecoder.java]: Replace the tab with spaces

**Review by aramperes — CHANGES_REQUESTED** [review]

**Review by aramperes — APPROVED** [review]

**Comment by aramperes** [comment]
Could you bump the library's version to `1.2.5-SNAPSHOT`?

**Comment by jdonkervliet** [comment]
Actually, I want to create a release for this repo. Can we first get to 1.2.4 before continuing with 1.2.5?

**Comment by aramperes** [comment]
Thanks for your contribution! I released 1.2.4 and started the 1.2.5 snapshots before merging this. This commit will be part of the 1.2.5 release at the end of the month.

**Comment by cedws** [comment]
We've discussed reverting this over [here](https://github.com/GlowstoneMC/Glowstone/pull/972). I think it might be a good idea to find out why this was implemented before reverting it. ~~We should also consider that reverting it will diverge further from upstream `flow/network`.~~ Looks like it wasn't upstreamed. Yeah, might as well revert it then.

**Pull request merged**